### PR TITLE
Add binary size local command shortcut to benchmark detail

### PR DESCRIPTION
**Pull request by Kobzol**
This PR adds a command shortcut for analyzing binary size locally using the new [`binary_stats`](https://github.com/rust-lang/rustc-perf/pull/1772) command to the benchmark detail panel.

image

### CODE REVIEW

File: site/frontend/src/pages/compare/compile/table/shortcuts/cachegrind-cmd.vue [hunk 1]
```diff
@@ -1,5 +1,8 @@
 <script setup lang="ts">
-import {CompileTestCase, Profile} from "../pages/compare/compile/common";
+/**
+ * This component displays a rustc-perf command for profiling a compile benchmark with Cachegrind.
+ * */
```
Comment (lqd) [inline site/frontend/src/pages/compare/compile/table/shortcuts/cachegrind-cmd.vue:4]: tiny nit: newline here, I thought this was describing the import 😓

File: site/frontend/src/pages/compare/compile/table/shortcuts/profile-shortcut.vue [hunk 2]
```diff
@@ -0,0 +1,82 @@
+<script setup lang="ts">
+import {computed, ref, Ref} from "vue";
```
Comment (lqd) [inline site/frontend/src/pages/compare/compile/table/shortcuts/profile-shortcut.vue:2]: If we're documenting the `CachegrindCmd` component above, then let's also do it for this shortcut file as well as the binary-size shortcut file.

File: site/frontend/src/pages/compare/compile/table/shortcuts/binary-size-shortcut.vue [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      `cargo build --release`, to compare binary sizes.
+    </Tooltip>
+  </div>
+
+  <pre><code>./target/release/collector binary_stats \
+    +{{ props.baseArtifact.commit }} \
+    --rustc2 +{{ props.artifact.commit }} \
+    --include {{ testCase.benchmark }} \
+    --profile {{ normalizeProfile(testCase.profile) }} \
+    --backend {{ normalizeBackend(testCase.backend) }}</code></pre>
```
Comment (lqd) [inline site/frontend/src/pages/compare/compile/table/shortcuts/binary-size-shortcut.vue:38]: Would we want the selectbox with "Diff/Baseline commit/Benchmarked commit" choices here, like for the profiling shortcut?

**Review by lqd — APPROVED** [review]
r=me with 2 tiny nits

File: site/frontend/src/pages/compare/compile/table/benchmark-detail.vue [hunk 4]
```diff
@@ -455,40 +433,20 @@ onMounted(() => {
...
-          Benchmarked commit
-        </option>
-      </select>
-
-      <CachegrindCmd
-        :commit="profileCommit"
-        :baseline-commit="profileBaselineCommit"
-        :test-case="testCase"
+    <div class="shortcut">
+      <template v-if="props.metric === BINARY_SIZE_METRIC">
```
Comment (lqd) [inline site/frontend/src/pages/compare/compile/table/benchmark-detail.vue:437]: (drive-by note that this check reminded me, would we also want to hide the "frontend/backend/linker" wall-time sections when displaying the size metrics?)

Comment (Kobzol) [reply site/frontend/src/pages/compare/compile/table/benchmark-detail.vue:437]: The FE/BE/Linker chart doesn't fit that well into the detail, since it's not metric-specific :/ But I wanted it to be easily discoverable. It really only makes sense currently for wall-time, or task-clock.

Maybe we could somehow highlight this fact under its title, like we do for the 30 day charts, where we explain what do they show.

File: site/frontend/src/pages/compare/compile/table/shortcuts/binary-size-shortcut.vue (hunk 3 above)
Comment (Kobzol) [reply site/frontend/src/pages/compare/compile/table/shortcuts/binary-size-shortcut.vue:38]: I decided not to add it, to avoid complicating the code. I think that in 99 % you want to see the diff anyway, and getting the binary stats should be much faster than running cachegrind, so I think that it's not worth it to run the command without the diff.

**Comment by Kobzol** [comment]
Added and expanded comments :+1:

File: site/frontend/src/pages/compare/compile/table/benchmark-detail.vue (hunk 4 above)
Comment (lqd) [reply site/frontend/src/pages/compare/compile/table/benchmark-detail.vue:437]: I'm finding a bit less helpful than I hoped it would be (as discussed in a recent regression);  we could make a bunch of improvements there. Hiding the chart on the size metrics seems like a straightforward one we probably should do (as it's very big)

File: site/frontend/src/pages/compare/compile/table/shortcuts/binary-size-shortcut.vue (hunk 3 above)
Comment (lqd) [reply site/frontend/src/pages/compare/compile/table/shortcuts/binary-size-shortcut.vue:43]: Seems fine. And it's also easy to understand once you're not a bigger with rustc-perf that `--rustc2` is the common pattern for diffs.

File: site/frontend/src/pages/compare/compile/table/benchmark-detail.vue (hunk 4 above)
Comment (Kobzol) [reply site/frontend/src/pages/compare/compile/table/benchmark-detail.vue:437]: Ok, I'll send a PR, right after this one gets merged.
